js/script.js

```diff
@@ -0,0 +1,145 @@
+        break;
+      case 'ended':
+          newGameElem.style.display = 'block';
+          newGameBtn.innerText = 'Jeszcze raz';
+          resultsElem.style.display = 'block';
+          pickElem.style.display = 'none';
+        break;
+      case 'notStarted':
+      default:
+          newGameElem.style.display = 'block';
+          pickElem.style.display = 'none';
+          resultsElem.style.display = 'none';
+    }
+}
+
+setGameElements();
+
+//Rozpoczęcie gry
+var playerPointsElem = document.getElementById('js-playerPoints');
+var playerNameElem = document.getElementById('js-playerName');
+var computerPointsElem = document.getElementById('js-computerPoints');
+
+function setGamePoints() {
+    playerPointsElem.innerHTML = player.score;
+    computerPointsElem.innerHTML = computer.score;
+}
+
+function newGame() {
+    player.name = prompt('Please enter your name', 'imię gracza');
+    if (player.name) {
+      player.score = computer.score = 0;
+
+      playerPickElem.innerText = "Player selection";
+      computerPickElem.innerText = "Computer selection";
+      playerResultElem.innerText = "Player Score";
+      computerResultElem.innerText = "Computer Score";
+
+      pickElem.style.display = 'block';
+      winnerAlertElem.innerText = "";
+
+      gameState = 'started';
+      setGameElements();
+  
+      playerNameElem.innerHTML = player.name;
+      setGamePoints();
+    }
+}
+
+//Wybór gracza
+function playerPick(playerPick) {
+    var computerPick = getComputerPick();
+    
+    playerPickElem.innerHTML = playerPick;
+    computerPickElem.innerHTML = computerPick;
+
+    checkRoundWinner(playerPick, computerPick);
+}
+
+//Wybór komputera
+function getComputerPick() {
+    var possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random()*3)];
+}
+
+//Wyświetlenie wyboru gracza i komputera na stronie
+var playerPickElem = document.getElementById('js-playerPick');
+var computerPickElem = document.getElementById('js-computerPick');
+var playerResultElem = document.getElementById('js-playerResult');
+var computerResultElem = document.getElementById('js-computerResult');
+
+//Sprawdzenie kto wygrał rundę
+function checkRoundWinner(playerPick, computerPick) {
+    playerResultElem.innerHTML = computerResultElem.innerHTML = '';
+  
+    var winnerIs = 'player';
+  
+      if (playerPick == computerPick) {
+          winnerIs = 'noone'; // remis
+      } else if (
+          (computerPick == 'rock' &&  playerPick == 'scissors') ||
+          (computerPick == 'scissors' &&  playerPick == 'paper') ||
+          (computerPick == 'paper' &&  playerPick == 'rock')) {
+  
+          winnerIs = 'computer';
+      }
+  
+      if (winnerIs == 'player') {
+          playerResultElem.innerHTML = "Win!";
+          player.score++;
+      } else if (winnerIs == 'computer') {
+          computerResultElem.innerHTML = "Win!";
+          computer.score++;
+      }
+    
+    setGamePoints();
+    gameOver();
+}
```

Comment:
Zauważ, że w tej funkcji robimy trochę masło maślane. Na podstawie picku gracza i komputera uzupełniamy zmienną `winnerIs`, a potem… sprawdzamy tą zmienną i zależnie od  jej wartości podejmujemy akcje.

Spróbuj przerobić to tak żebyśmy zamiast wypełniać zmienną `winnerIs` - od razu podejmowali odpowiednie akcje. Te   akcje najlepiej ubrać w nową funkcję bo w obecym stanie rzeczy ona robi zbyt wiele rzeczy na raz. Teoretycznie funkcja powinna robić maksymalnie jedną rzecz (zasada single responsibility)